most_free_time.rb

```diff
@@ -13,8 +13,9 @@
 =end
 
 # my solution seems passed all given tests, accoridng to the return values. 
-# but doesn't show in the system. 
-# need code review and refactoring???????????????????????
+# but in coderbyte, it didn't show i passed all tests. 
```

Comment:
@bhennes2 just saw your note, my failed tests as below: 
1. When the input was ("10:00AM-12:30PM","02:00PM-02:45PM","09:10AM-09:50AM") your output was incorrect.
2. When the input was ("12:15PM-02:00PM","09:00AM-10:00AM","10:30AM-12:00PM") your output was incorrect.
3. When the input was ("12:15PM-02:00PM","09:00AM-12:11PM","02:02PM-04:00PM") your output was incorrect.
4. When the input was ("07:00AM-08:00AM","09:00AM-10:00AM","10:00PM-11:00PM") your output was incorrect.
5. When the input was ("07:00AM-08:00AM","10:00PM-11:00PM","09:00AM-09:15PM") your output was incorrect.
6. When the input was ("01:52AM-01:55AM","01:00AM-01:10AM","01:20AM-01:50AM") your output was incorrect.
7. When the input was ("11:30AM-01:00PM","07:00AM-08:15AM","01:50PM-05:00PM","08:30AM-11:10AM") your output was incorrect.
8. When the input was ("11:30AM-01:00PM","07:00AM-08:15AM","01:05PM-05:00PM","08:30AM-11:10AM") your output was incorrect.
9. When the input was ("06:00AM-08:00PM","09:09PM-09:11PM","08:02PM-08:04PM","08:10PM-09:00PM") your output was incorrect.
10. When the input was ("06:00AM-08:00PM","05:00AM-05:45AM","08:02PM-08:04PM","08:10PM-09:00PM","09:09PM-09:11PM") your output was incorrect.